lib/kafka_gen_stage/consumer_logic.ex

```diff
@@ -28,71 +27,78 @@ defmodule KafkaGenStage.ConsumerLogic do
   @spec prepare_dispatch(
           buffered_msgs :: :queue.queue(),
           buffered_demand :: non_neg_integer(),
-          transformer :: (msg_tuple(), term() -> {term(), term()}),
-          transformer_state :: term,
-          bulk_transformer :: ([msg_tuple()] -> [msg_tuple()])
+          bulk_transformer :: ([msg_tuple()] -> [msg_tuple()]),
+          high_wm :: non_neg_integer()
```

Comment:
I think this type is incorrect. If a consumer asks for demand sooner than we get anything from kafka, then this function is called with `:unknown`. I'm not entirely sure how likely that is to happen, so please correct me if I'm wrong.

Reply:
(You do seem to handle this case in transform_bulk/3 so I guess it does happen.)

Reply:
Exactly, it does not happen.

Reply:
Why do handle it then?

Reply:
It does happen exactly in the case that is handled in `transform_bulk/3` by not calling the bulk_transformer fun.